### ISSUE TITLE
Cannot retrieve caches by location

### ISSUE THREAD

**Issue by samueltardieu** [cgeo/cgeo#4616]
(initially reported from Jan. 14 on support in French)

Problem: version 2015.01.12, "nearby caches" never terminates.

Settings:
- basic member;
- WiFi and 3G;
- login ok;
- fresh installation doesn't work either;
- 2014.12.22 works [exact version to be confirmed].

Information (seems copied by hand):

```
device : GT-S7390G ( kylevessxw, samsung)
android version : 4.12
android build : JZ054.S7390GXXUANJ1
cgeo version 2015.01.12
play service : present 
use play service : no
compas capability : no 
rotation sensor : abscent
geomagnetic rotation sensor : abscent
orientation sensors : abscent
```

**Comment by SammysHP** [comment]
> abscent

?!

**Comment by samueltardieu** [comment]
Another report with a never terminating "nearby caches".

Info OCR-ed from screenshot sent by user:

```
--- System information ---
Device: GT-I8190 (goldenxx, samsung)
Android version: 4.1.2
Android build: JZ054K.18190XXANA2
Cgeo version: 2015.01.12
Play services: present
Use Play services: yes
Low power mode: inactive
Compass capabilities: yes
Rotation sensor: present
Geomagnetic rotation sensor: absent
Orientation sensor: present
--- End of system information --- 
```

**Comment by Lineflyer** [comment]
Another german user on support mail:

```
--- System information ---
Device: ST17i (ST17i_1254-2724, SEMC)
Android version: 2.3.4
Android build: 4.0.2.A.0.84
Cgeo version: 2015.01.12
Play services: absent
Low power mode: inactive
Compass capabilities: yes
Rotation sensor: present
Geomagnetic rotation sensor: absent
Orientation sensor: present
--- End of system information ---
```

**Comment by Lineflyer** [comment]
And another:

```
--- System information ---
Device: GT-S5830i (GT-S5830i, samsung)
Android version: 2.3.6
Android build: GINGERBREAD.XXMJ2
Cgeo version: 2015.01.12
Play services: absent
Low power mode: inactive
Compass capabilities: yes
Rotation sensor: present
Geomagnetic rotation sensor: absent
Orientation sensor: present
--- End of system information ---
```

**Comment by samueltardieu** [comment]
@Lineflyer Please get logs from those Android 2.x users, as well as logs from them running 2014.12.22 (and confirmation that it works) so that we can compare what is different between the working and the non-working versions.

**Comment by Lineflyer** [comment]
Yup, requested. Additionally I requested the user status (BM/PM) although we should probably see it in the log.

BTW: This brings to to some idea for additional system info lines. Will add it in a separate issue.

**Comment by Lineflyer** [comment]
Working for me with 2015.01.12 as BM and PM.

Was also working for me while doing regression testing before the release.

**Comment by rsudev** [comment]
Also working for me - quite mysterious.

**Comment by Lineflyer** [comment]
Sadly we wont get a log from the 2.3.4 user, as he answered that he is unable to install a log application due to memory restrictions (and he is not willing to deinstall other apps for that).
But he at least confirmed that 2014.12.22 is working again for him.

**Comment by samueltardieu** [comment]
Won't he install adb on his machine and get the logs via:

`adb logcat -v time -s cgeo AndroidRuntime:e`

?

**Comment by Lineflyer** [comment]
I just send a log to the dev mailing list. DIdn't take a look myself yet.

If I crawled the support mails correct, then all of the affected users are basic members.
Could it have something to do with the ANR for directional image loading?

**Comment by rsudev** [comment]
A first look doesn't show any apparent difference between the working and the non-working version. Especially the search requests are successfully answered in both cases, and caches are loaded from the database. There are no requests for direction images visible at first sight...

**Comment by rsudev** [comment]
It is always good to write such a not, than you have something to take back again when you take another look...
The user has three providers enabled, gc.com, oc.de and oc.com. In the error case, it seems that only the answer from oc.com is processed, no activity on gc.com/oc.de caches visible in the log there...

**Comment by Lineflyer** [comment]
I was able to reproduce a same behaviour with a basic member account of a friend. It does never happen with my own BM account, but with his BM account on the same device it happens from time to time.

I tried four times the same search by coords. The fourth try fails...It keeps showing the spinning circle in header and centered on the screen for at least 10 minutes.

This is the "cgeo" filtered log:
```
22:58:30.000 Debug cgeo 21048  [looper callbacks] OrientationProvider: stopping the orientation provider
22:58:33.822 Info cgeo 21506  [main] Google Play services are available
22:58:33.822 Debug cgeo 21506  [main] OrientationProvider: sensor found
22:58:33.822 Debug cgeo 21506  [network-1] GET http://status.cgeo.org/api/status.json?version_code=20150112&version_name=2015.01.12&locale=de_DE
22:58:33.932 Info cgeo 21506  [main] Starting cgeo.geocaching 20150112 a.k.a 2015.01.12
22:58:33.932 Debug cgeo 21506  [RxCachedThreadScheduler-2] Database clean: started
22:58:33.932 Debug cgeo 21506  [RxCachedThreadScheduler-2] Database clean: removing 0 geocaches from listId=0
22:58:33.932 Debug cgeo 21506  [looper callbacks] GpsStatusProvider: starting the GPS status listener
22:58:33.932 Debug cgeo 21506  [RxCachedThreadScheduler-2] Database clean: removing obsolete log images records
22:58:33.932 Debug cgeo 21506  [RxCachedThreadScheduler-2] Database clean: finished
22:58:33.932 Debug cgeo 21506  [network-2] GET https://www.geocaching.com/login/default.aspx
22:58:34.041 Debug cgeo 21506  [main] LocationProvider: requesting most precise locations
22:58:34.259 Debug cgeo 21506  [network-1] 200 (361 ms) GET http://status.cgeo.org/api/status.json?version_code=20150112&version_name=2015.01.12&locale=de_DE
22:58:35.024 Debug cgeo 21506  [network-2] 200 (1106 ms) GET https://www.geocaching.com/login/default.aspx
22:58:35.024 Info cgeo 21506  [network-2] Already logged in Geocaching.com as xxx (Basic Member)
22:58:35.024 Info cgeo 21506  [network-2] Geocaching.com language already set to English
22:58:35.133 Debug cgeo 21506  [network-2] GET http://www.geocaching.com/email/
22:58:35.788 Debug cgeo 21506  [network-2] 200 (665 ms) GET http://www.geocaching.com/email/
22:58:36.006 Debug cgeo 21506  [network-5] GET https://www.geocaching.com/account/settings/homelocation
22:58:36.006 Debug cgeo 21506  [network-2] GET https://www.geocaching.com/account/settings/preferences
22:58:37.208 Debug cgeo 21506  [network-2] 200 (1203 ms) GET https://www.geocaching.com/account/settings/preferences
22:58:37.208 Debug cgeo 21506  [network-5] 200 (1217 ms) GET https://www.geocaching.com/account/settings/homelocation
22:58:37.317 Info cgeo 21506  [network-5] Setting home location to N 52° 2x.xxx′ E 13° 3x.xxx′
22:58:39.392 Debug cgeo 21506  [looper callbacks] GpsStatusProvider: stopping the GPS status listener
22:58:41.903 Debug cgeo 21506  [looper callbacks] LocationProvider: stopping location requests
22:58:49.220 Debug cgeo 21506  [main] LocationProvider: requesting most precise locations
22:58:49.220 Debug cgeo 21506  [looper callbacks] OrientationProvider: starting the orientation provider
22:58:49.220 Debug cgeo 21506  [network-6] GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:58:50.093 Debug cgeo 21506  [network-6] 200 (822 ms) GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:58:50.530 Debug cgeo 21506  [network-6] DataStore.loadCaches([GC5CDYT, GC544KD, GC58A9N, GC57F79, GC533N5, GC5FEQW, GC53FMX, GC3NXVZ, GC544M6, GC4W4T6, GC5414E, GC537EV, GC2EMJ4, GCXZRY, GC4WC94, GC544MW, GC5BRM7, GC53FM8, GC544J7, GC58AA0]) returned no results
22:58:50.639 Debug cgeo 21506  [network-6] GET http://gcvote.com/getVotes.php?version=cgeo&waypoints=GC5414E%2CGC537EV%2CGC2EMJ4%2CGC5CDYT%2CGC544KD%2CGCXZRY%2CGC4WC94%2CGC58A9N%2CGC57F79%2CGC544MW%2CGC5BRM7%2CGC53FM8%2CGC533N5%2CGC5FEQW%2CGC53FMX%2CGC3NXVZ%2CGC544M6%2CGC4W4T6%2CGC58AA0%2CGC544J7
22:58:50.967 Debug cgeo 21506  [network-6] 200 (311 ms) GET http://gcvote.com/getVotes.php?version=cgeo&waypoints=GC5414E%2CGC537EV%2CGC2EMJ4%2CGC5CDYT%2CGC544KD%2CGCXZRY%2CGC4WC94%2CGC58A9N%2CGC57F79%2CGC544MW%2CGC5BRM7%2CGC53FM8%2CGC533N5%2CGC5FEQW%2CGC53FMX%2CGC3NXVZ%2CGC544M6%2CGC4W4T6%2CGC58AA0%2CGC544J7
22:58:53.369 Debug cgeo 21506  [looper callbacks] OrientationProvider: stopping the orientation provider
22:58:54.570 Debug cgeo 21506  [looper callbacks] OrientationProvider: starting the orientation provider
22:58:54.570 Debug cgeo 21506  [network-7] GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:58:55.226 Debug cgeo 21506  [network-7] 200 (711 ms) GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:58:55.772 Debug cgeo 21506  [network-7] GET http://gcvote.com/getVotes.php?version=cgeo&waypoints=GC5414E%2CGC537EV%2CGC2EMJ4%2CGC5CDYT%2CGC544KD%2CGCXZRY%2CGC4WC94%2CGC58A9N%2CGC57F79%2CGC544MW%2CGC5BRM7%2CGC53FM8%2CGC533N5%2CGC5FEQW%2CGC53FMX%2CGC3NXVZ%2CGC544M6%2CGC4W4T6%2CGC58AA0%2CGC544J7
22:58:56.208 Debug cgeo 21506  [network-7] 200 (352 ms) GET http://gcvote.com/getVotes.php?version=cgeo&waypoints=GC5414E%2CGC537EV%2CGC2EMJ4%2CGC5CDYT%2CGC544KD%2CGCXZRY%2CGC4WC94%2CGC58A9N%2CGC57F79%2CGC544MW%2CGC5BRM7%2CGC53FM8%2CGC533N5%2CGC5FEQW%2CGC53FMX%2CGC3NXVZ%2CGC544M6%2CGC4W4T6%2CGC58AA0%2CGC544J7
22:58:58.392 Debug cgeo 21506  [looper callbacks] OrientationProvider: stopping the orientation provider
22:58:59.157 Debug cgeo 21506  [looper callbacks] OrientationProvider: starting the orientation provider
22:58:59.157 Debug cgeo 21506  [network-8] GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:58:59.921 Debug cgeo 21506  [network-8] 200 (749 ms) GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:59:00.467 Debug cgeo 21506  [network-8] GET http://gcvote.com/getVotes.php?version=cgeo&waypoints=GC5414E%2CGC537EV%2CGC2EMJ4%2CGC5CDYT%2CGC544KD%2CGCXZRY%2CGC4WC94%2CGC58A9N%2CGC57F79%2CGC544MW%2CGC5BRM7%2CGC53FM8%2CGC533N5%2CGC5FEQW%2CGC53FMX%2CGC3NXVZ%2CGC544M6%2CGC4W4T6%2CGC58AA0%2CGC544J7
22:59:00.686 Debug cgeo 21506  [network-8] 200 (269 ms) GET http://gcvote.com/getVotes.php?version=cgeo&waypoints=GC5414E%2CGC537EV%2CGC2EMJ4%2CGC5CDYT%2CGC544KD%2CGCXZRY%2CGC4WC94%2CGC58A9N%2CGC57F79%2CGC544MW%2CGC5BRM7%2CGC53FM8%2CGC533N5%2CGC5FEQW%2CGC53FMX%2CGC3NXVZ%2CGC544M6%2CGC4W4T6%2CGC58AA0%2CGC544J7
22:59:02.760 Debug cgeo 21506  [looper callbacks] OrientationProvider: stopping the orientation provider
22:59:04.071 Debug cgeo 21506  [looper callbacks] OrientationProvider: starting the orientation provider
22:59:04.071 Debug cgeo 21506  [network-9] GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
22:59:04.835 Debug cgeo 21506  [network-9] 200 (682 ms) GET http://www.geocaching.com/seek/nearest.aspx?lat=51.34765&lng=4.044383333333333&tx=9a79e6ce-3344-409c-bbe9-496530baf758
```

Briefly looking over the complete log, I saw this happening in parallel to sending of the last request:
```
22:59:04.071 Error Parcel 2715  Class not found when unmarshalling: cgeo.geocaching.location.Geopoint
22:59:04.071 Error Parcel 2715  java.lang.ClassNotFoundException: cgeo.geocaching.location.Geopoint
22:59:04.071 Error Parcel 2715  	at java.lang.Class.classForName(Native Method)
22:59:04.071 Error Parcel 2715  	at java.lang.Class.forName(Class.java:251)
22:59:04.071 Error Parcel 2715  	at java.lang.Class.forName(Class.java:216)
22:59:04.071 Error Parcel 2715  	at android.os.Parcel.readParcelableCreator(Parcel.java:2133)
22:59:04.071 Error Parcel 2715  	at android.os.Parcel.readParcelable(Parcel.java:2097)
22:59:04.071 Error Parcel 2715  	at android.os.Parcel.readValue(Parcel.java:2013)
22:59:04.071 Error Parcel 2715  	at android.os.Parcel.readArrayMapInternal(Parcel.java:2314)
22:59:04.071 Error Parcel 2715  	at android.os.Bundle.unparcel(Bundle.java:249)
22:59:04.071 Error Parcel 2715  	at android.os.Bundle.getString(Bundle.java:1118)
22:59:04.071 Error Parcel 2715  	at android.content.Intent.getStringExtra(Intent.java:5128)
22:59:04.071 Error Parcel 2715  	at com.android.server.am.ActivityStackSupervisor.startActivityLocked(ActivityStackSupervisor.java:1447)
22:59:04.071 Error Parcel 2715  	at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:1043)
22:59:04.071 Error Parcel 2715  	at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:4054)
22:59:04.071 Error Parcel 2715  	at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:3948)
22:59:04.071 Error Parcel 2715  	at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:159)
22:59:04.071 Error Parcel 2715  	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2629)
22:59:04.071 Error Parcel 2715  	at android.os.Binder.execTransact(Binder.java:404)
22:59:04.071 Error Parcel 2715  	at dalvik.system.NativeStart.run(Native Method)
22:59:04.071 Error Parcel 2715  Caused by: java.lang.NoClassDefFoundError: cgeo/geocaching/location/Geopoint
22:59:04.071 Error Parcel 2715  	... 18 more
22:59:04.071 Error Parcel 2715  Caused by: java.lang.ClassNotFoundException: Didn't find class "cgeo.geocaching.location.Geopoint" on path: DexPathList[[directory "."],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
22:59:04.071 Error Parcel 2715  	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:67)
22:59:04.071 Error Parcel 2715  	at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
22:59:04.071 Error Parcel 2715  	at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
22:59:04.071 Error Parcel 2715  	... 18 more
22:59:04.071 Error dalvikvm 2715  dvmPauseGc(GestureDetector) called - cookie=0xfac (f=0x1)
```

Full log is saved here, but have no more time today.

**Comment by Lineflyer** [comment]
Another user:

```
--- System information ---
Device: MB526 (MB526_RTDACH, MOTO)
Android version: 2.3.6
Android build: 4.5.1-134_DFP-231
Cgeo version: 2015.01.12
Play services: present
Use Play services: no
Low power mode: inactive
Compass capabilities: yes
Rotation sensor: present
Geomagnetic rotation sensor: absent
Orientation sensor: present
--- End of system information ---
```
Debug log requested

**Comment by Lineflyer** [comment]
New mails on support mail are varying:
- Some users who reported it in the past, now report that it is working again
- Some new users in parallel where it does not work
- One user tells that with the same account it is working on one device but not on another device of the same type

**Comment by Lineflyer** [comment]
Another log from myself with BM account on an Android 2.3.5 device:

```
21:09:46.782 Debug cgeo 1263  [network-6] GET https://www.geocaching.com/account/settings/preferences
21:09:46.892 Debug cgeo 1263  [network-7] 200 (8511 ms) POST https://www.geocaching.com/login/default.aspx
21:09:47.212 Info cgeo 1263  [network-7] Geocaching.com language already set to English
21:09:47.212 Debug cgeo 1263  [network-7] GET http://www.geocaching.com/email/
21:09:51.582 Debug cgeo 1263  [network-7] 200 (4342 ms) GET http://www.geocaching.com/email/
21:09:52.022 Debug cgeo 1263  [network-6] 200 (5321 ms) GET https://www.geocaching.com/account/settings/preferences
21:09:52.022 Info cgeo 1263  [network-7] Successfully logged in Geocaching.com as xxxxxxxxx (Basic Member)
21:09:52.132 Debug cgeo 1263  [network-7] GET https://www.geocaching.com/account/settings/preferences
21:09:54.532 Debug cgeo 1263  [network-7] 200 (2482 ms) GET https://www.geocaching.com/account/settings/preferences
21:09:55.622 Debug cgeo 1263  [looper callbacks] GpsStatusProvider: stopping the GPS status listener
21:09:55.732 Debug cgeo 1263  [looper callbacks] OrientationProvider: starting the orientation provider
21:09:55.842 Debug cgeo 1263  [network-8] GET http://www.geocaching.com/seek/nearest.aspx?lat=51.xxx&lng=7.xxx8&tx=9a79e6ce-3344-409c-bbe9-496530baf758
21:09:59.992 Debug cgeo 1263  [network-8] 200 (4121 ms) GET http://www.geocaching.com/seek/nearest.aspx?lat=51.xxx&lng=7.xxx&tx=9a79e6ce-3344-409c-bbe9-496530baf758
21:10:00.972 Debug cgeo 1263  [network-8] DataStore.loadCaches([GC5HQ4B, GC4EQ2N, GC352Y3, GC3MCKK, GC5HQ4J, GC533KT, GC5HQ47, GC150HM, GC2DZXV, GC5HQ40, GC40D80, GC4N13Z, GC5GVK0, GC3EQ4W, GC34M2J, GC3KGEJ, GC4ZCRJ, GC5HQ4R, GC5HQ2Z, GC588BF]) returned no results
```
After that nothing is happening (waited at least 10 minutes).

**Comment by Lineflyer** [comment]
OK...after an evening full of testing with different devices, accounts, versions these are my (perhaps totally incorrect) conclusions:
- It happend only for Basic Members, PM was OK all the time
- It does not always happen for BM but seems to depend on something unkown
- My personal impression is that is is more likely to happen on slower devices or devices with Android 2.3.x (but also happens fom time to time on newest devices)

And most of all:
- It never happend for me using 2014.12.22 or 2015.01.18-NB but only on 2015.01.12 !!

and:
In the logs for my tests with 2015.01.12 using a BM-account I did **never** see outgoing requests for direction images like this:
```
21:30:03.821 Debug cgeo 1679  [network-9] GET https://www.geocaching.com/images/icons/compass/N.gif
```
However the images are still shown in case the nearby search was OK. So I am unsure whether this debug info was only add in the new nightlies or the change introduced by @samueltardieu some days ago changed something (which implicitly fixed the problem).

**Comment by samueltardieu** [comment]
Is it possible that once the images are there (i.e., loaded with a version which works), they are used because they don't need to be downloaded anymore? And indeed I changed a problem with the loading of those images in the latest version. We should ask people having problems to try the -NB (or build a new -RC to let them try it).

**Comment by Lineflyer** [comment]
Or we could ask the users to deactivate the directional image loading (which I did not test yet), but this would only proof that it is the culprit.
So indeed a new RC (for reasons of an easy upgrade path for the users) would be IMO the best approach. Its a little try and error but it it the best we have.

It would be interesting to learn why the error is only showing up in 2015.01.12 but not in 2014.12.22. But still....RC would be worth a try.

**Comment by Lineflyer** [comment]
Despite the current problems in sending support mail, two of my replies to affected users have gone through and we received the feedback, that it solved the problem.
 
We should release as soon as reasonable.

**Comment by Lineflyer** [comment]
As the mail system is working again, there is more positive feedback.
So we can be quite sure, that the problem is solved on `release` branch.
